homeassistant/components/hassio/discovery.py

```diff
@@ -0,0 +1,140 @@
+"""Implement the serivces discovery feature from Hass.io for Add-ons."""
+import asyncio
+import logging
+
+from aiohttp import web
+from aiohttp.web_exceptions import HTTPServiceUnavailable
+
+from homeassistant.core import callback
+from homeassistant.const import EVENT_HOMEASSISTANT_START
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.helpers.discovery import async_load_platform, async_discover
+
+from .handler import HassioAPIError
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_DISCOVERY = 'discovery'
+ATTR_ADDON = 'addon'
+ATTR_NAME = 'name'
+ATTR_SERVICE = 'service'
+ATTR_CONFIG = 'config'
+ATTR_COMPONENT = 'component'
+ATTR_PLATFORM = 'platform'
+ATTR_UUID = 'uuid'
+
+CONFIG_FLOW_SERVICE = ('mqtt',)
+
+
+@callback
+def async_setup_discovery(hass, hassio, config):
+    """Discovery setup."""
+    hassio_discovery = HassIODiscovery(hass, hassio, config)
+
+    # Handle exists discovery messages
+    async def async_discovery_start_handler(event):
+        """Process all exists discovery on startup."""
+        try:
+            data = await hassio.retrieve_discovery_messages()
+        except HassioAPIError as err:
+            _LOGGER.error("Can't read discover info: %s", err)
+            return
+
+        jobs = [hassio_discovery.async_process_new(discovery)
+                for discovery in data[ATTR_DISCOVERY]]
+        if jobs:
+            await asyncio.wait(jobs)
+
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_START, async_discovery_start_handler)
+
+    hass.http.register_view(hassio_discovery)
+
+
+class HassIODiscovery(HomeAssistantView):
+    """Hass.io view to handle base part."""
+
+    name = "api:hassio_push:discovery"
+    url = "/api/hassio_push/discovery/{uuid}"
+
+    def __init__(self, hass, hassio, config):
+        """Initialize WebView."""
+        self.hass = hass
+        self.hassio = hassio
+        self.config = config
+
+    async def post(self, request):
```

Comment:
Because you have `{uuid}` in the url, it will be passed as extra param to this method.